### PR TITLE
Mark tests

### PR DESCRIPTION
**Pull request by parikshit14**
@davidt0x updated the branch as discussed. 
- Though I didn't change `setup.py` for `[test]` because I think it should remain in the dev installation as to test locally one doesn't need to install the entire list of optional packages.
- As discussed with @pgleeson, we wanted to have `PyTorch` as core dependencies as we provide specifically high functionalities and examples for it, so we might want to think about that.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -39,22 +39,16 @@ install_requires =
...
     onnxruntime
     skl2onnx
     attrs
     cattrs
-    neuromllite>=0.5.1
     modelspec>=0.2.4
-    pyNeuroML>=0.5.20
     h5py
-    psyneulink>=0.12.0.0
     h5py
```
Comment (davidt0x) [inline setup.cfg:50]: Remove this copy of h5py

File: setup.py [hunk 2]
```diff
@@ -23,6 +23,12 @@
         "pytorch-sphinx-theme==0.0.19",
         "sphinxcontrib-versioning",
         "Jinja2<3.1",
+        "neuromllite>=0.5.1",
+        "pyNeuroML>=0.5.20",
```
Comment (davidt0x) [inline setup.py:27]: I think these should be broken apart into individual extra requirements for each backend, Something like:
```
extras = {
    "psyneulink": ["psyneulink>=0.12.0.0"],
    "neuroml": ["pyNeuroML>=0.5.20", "neuromllite>=0.5.1"],
 ...

```       

We can then have `optional` be all core plus all backed requirements.

**Review by davidt0x — CHANGES_REQUESTED** [review]
I think pytorch shouldn't be a core requirement if we can avoid it but I will discuss with @pgleeson tomorrow in our meeting.

**Comment by davidt0x** [comment]
Note sure what is causing the CI fails on linux and windows. Might be tricky to debug, looks like an out of memory error maybe.

**Comment by parikshit14** [comment]
> Note sure what is causing the CI fails on linux and windows. Might be tricky to debug, looks like an out of memory error maybe.

yes, I ran the "not core" tests locally and it was crashing there as well after few minutes.

**Comment by davidt0x** [comment]
I think this is being because modelspec 0.2.4 is being installed in place of the github repo version when optional dependencies are installed. There was a bug fix after 0.2.4 (not sure if its in 0.2.5 or later). The bug was that the evaluate function was causing all the tensors to be formatted to strings regardless of whether they were actually printing or not. Anyway, the root cause is that somewhere pip thinks it needs 0.2.4 when in should be pinned higher.

**Comment by pgleeson** [comment]
@parikshit14 Try pulling in the latest changes from the development branch to yours, some of the issues with getting the correct version of modelspec/psyneulink are resolved there.

**Comment by pgleeson** [comment]
@parikshit14 This PR contains a lot of extra files related to the updates for integrating the readmes into the docs. These have already been selectively integrated into the docs (many image/py/json files are not yet required), so could you open a fresh PR with just the changes for marking the tests (ci.yml, setup.py etc)?

**Pull request closed without merging**